integrations/access/pagerduty/testlib/suite.go

```diff
@@ -453,24 +453,27 @@ func (s *PagerdutySuiteOSS) TestRecipientsFromAccessMonitoringRule() {
 		})
 	assert.NoError(t, err)
 
-	// Test execution: create an access request
-	req := s.CreateAccessRequest(ctx, integration.RequesterOSSUserName, nil)
-
-	// Validate the incident has been created in Pagerduty and its ID is stored
-	// in the plugin_data.
-	pluginData := s.checkPluginData(ctx, req.GetName(), func(data pagerduty.PluginData) bool {
-		return data.IncidentID != ""
-	})
+	// Incident creation may happen before plugins Access Monitoring Rule cache
+	// has been updated with new rule. Retry until the new rule starts applying.
+	require.Eventually(t, func() bool {
+		// Test execution: create an access request
+		req := s.CreateAccessRequest(ctx, integration.RequesterOSSUserName, nil)
```

Comment:
should this run in the eventually clause?

Reply:
Shouldn't you create **one** access request and continuously check until the AMR cache picks it up and starts to apply it?

Reply:
Yeah, I think so otherwise you are just creating 10 access requests that you immediately check if the rules were applied which will cause the flakiness to continue. 
What you should do instead is create the access request outside of the eventually loop and in the eventually loop check if the rules were correctly applied

Reply:
I was trying to avoid adding test specific code to the access monitoring rule watcher/cache to allow for this watching. If we create one request and it gets processed before the rule is added to the cache no amount of waiting will cause it to change recipients since they are determined as soon as the request is seen by the plugin. I could add a callback to the plugins or to the watcher to allow the tests to know if the rule has been picked up before creating the request, but that would require testing specific code being added outside of the _test.go files.

Reply:
I can add in the test specific code to the access monitoring rule cache so we can wait until its updated if you think that is better than just retrying the request creation until it catches up but having no test specific code leaking into the implementation.

Reply:
Is something like this more along the lines of what you were thinking?